src/text-expander-element.ts

```diff
@@ -189,8 +189,10 @@ class TextExpander {
   onKeydown(event: KeyboardEvent) {
     if (event.key !== 'Escape') return
     this.deactivate()
-    event.stopImmediatePropagation()
-    event.preventDefault()
+    if (this.menu || this.combobox) {
```

Comment:
Won't `this.deactivate()` reset those properties? It may always be false. I think we should capture this condition in some const before calling `this.deactivate()`.

Reply:
i moved the if check up.